pallets/settlement/src/lib.rs

```diff
@@ -994,6 +1001,45 @@ decl_module! {
             })
         }
 
+        /// Manually execute settlement
+        ///
+        /// # Arguments
+        /// * `id` - Target instruction id to reschedule.
+        /// * `_legs_count` - Legs included in this instruction.
+        ///
+        /// # Errors
+        /// * `InstructionNotFailed` - Instruction not in a failed state or does not exist.
+        #[weight = <T as Config>::WeightInfo::execute_manual_instruction(*legs_count)]
+        pub fn execute_manual_instruction(origin, id: InstructionId, legs_count: u32, portfolio: Option<PortfolioId>) {
+            // check origin has the permissions required and valid instruction
+            let (did, sk, instruction_details) = Self::ensure_origin_perm_and_instruction_validity(origin, id, true)?;
+
+            // Check for portfolio
+            match portfolio {
+                Some(portfolio) => {
+                    // Ensure that the sender is a party of this instruction.
+                    T::Portfolio::ensure_portfolio_custody_and_permission(portfolio, did, sk.as_ref())?;
+                    let mut legs = InstructionLegs::iter_prefix(id);
+                    ensure!(
+                        legs.any(|(_, leg)| [leg.from, leg.to].contains(&portfolio)),
+                        Error::<T>::UnauthorizedSigner
+                    );
+                }
+                None => {
+                    // Ensure venue exists & sender is its creator.
+                    Self::venue_for_management(instruction_details.venue_id, did)?;
+                }
+            }
+
+            // check that the instruction leg count matches
+            ensure!(InstructionLegs::iter_prefix(id).count() as u32 == legs_count, Error::<T>::LegCountTooSmall);
```

Comment:
I think we can allow a higher `legs_count` value.  The caller will just pay a higher fee if they hard-code the maximum 10 legs.  It is also what we do in `reject_instruction` extrinsic.
```suggestion
            ensure!(InstructionLegs::iter_prefix(id).count() as u32 <= legs_count, Error::<T>::LegCountTooSmall);
```

Reply:
Also we should try to avoid reading the `InstructionLegs` storage twice here.  It would also be good to pass the `legs` and `instruction_details` into `execute_instruction_retryable` to improve the performance.  But not in this PR.